### PR TITLE
Update Codecov upload API to v4

### PR DESCRIPTION
**Pull request by ararslan**
Guessing based on https://docs.codecov.io/reference#upload.

### CODE REVIEW

**Comment by ararslan** [comment]
RIP, looks like there's no CI here anymore. Guess we'll need to set up GHA.

File: src/codecovio.jl [hunk 1]
```diff
@@ -222,11 +222,15 @@ module Codecov
...
         if !dry_run
-            heads   = Dict("Content-Type" => "application/json",
-                           "Accept" => "application/json")
-            data    = to_json(fcs)
-            req     = HTTP.post(uri_str; body = JSON.json(data), headers = heads)
-            @debug "Result of submission:" * String(req)
+            # Tell Codecov we have an upload for them
+            response = HTTP.post(uri_str; headers=Dict("Accept" => "text/plain"))
+            # Get the temporary URL to use for uploading to S3
+            s3url = split(String(response.body), '\n')[2]
```
Comment (vtjnash) [inline src/codecovio.jl:228]:
```suggestion
            s3url = get(split(String(response.body), '\n'), 2, "")
            @debug "Result of submission:" * chomp(replace(String(req), s3url => ""))
            startswith(s3url, "https://") || error("Invalid codecov response: $s3url")
```

Comment (ararslan) [reply src/codecovio.jl:228]: `req` doesn't exist here, I think you meant `response`? Also, won't `String(response.body)` take ownership of the body and leave `String(response)` not informative, or does the `String` method show headers and whatnot?

Comment (vtjnash) [reply src/codecovio.jl:228]: ah, true, it does add headers, but the other call does destroy the body
```suggestion
            repr = String(response)
            s3url = get(split(String(response.body), '\n'), 2, "")
            repr = chomp(replace(repr, s3url => ""))
            @debug "Result of submission:" * repr
            startswith(s3url, "https://") || error("Invalid codecov response: $s3url")
```

**Comment by vtjnash** [comment]
Can you rebase and re-enable CI submission on https://github.com/JuliaCI/Coverage.jl/blob/master/etc/travis-coverage.jl#L4

**Comment by vtjnash** [comment]
You'll need to remove the conditional on that line. Looks like you might need to disable the blackhole server that we use for testing also (or provide a way to specify the API via an environment variable, and use v2 there?)

**Comment by DilumAluthge** [comment]
I have rebased, and I have enabled submission (i.e. we've removed all conditionals from https://github.com/JuliaCI/Coverage.jl/blob/master/etc/travis-coverage.jl), but it looks like CI is still failing.

**Comment by vtjnash** [comment]
Yeah, it fails on the blackhole server we used for testing.

**Comment by DilumAluthge** [comment]
@ararslan @vtjnash Based on #316, it seems like the only issue with this PR is the blackhole server bug. How do y'all want to proceed?

**Comment by ararslan** [comment]
¯\\\_(ツ)\_/¯

**Comment by DilumAluthge** [comment]
@vtjnash Can you explain the purpose of the black-hole server?

**Comment by vtjnash** [comment]
It avoided cluttering the list of log files with duplicate entries–no practical benefit to it

**Comment by DilumAluthge** [comment]
Alright, I think I have a way to fix it. Basically:
1. If we are not using the "black hole" server, we upload to S3
2. If we are using the "black hole" server, we do not upload to S3

See PR #318.

For the diff between this PR and PR #318, see here: https://github.com/JuliaCI/Coverage.jl/compare/aa/codecov-v4...dpa/codecov-v4

**Pull request closed without merging**